Xamarin.Forms.Platform.Android/Renderers/ListViewRenderer.cs

```diff
@@ -284,13 +284,18 @@ void OnScrollToRequested(object sender, ScrollToRequestedEventArgs e)
 			if (Element.IsGroupingEnabled)
 			{
 				var results = templatedItems.GetGroupAndIndexOfItem(scrollArgs.Group, scrollArgs.Item);
-				if (results.Item1 == -1 || results.Item2 == -1)
+				if (results.Item1 == -1 || (results.Item1 == -1 && results.Item2 == -1))
```

Comment:
Should this be
if (results.Item1 == -1 || (results.Item1 **!=** -1 && results.Item2 == -1))
The way it is currently, the second condition is irrelevant

Reply:
@nickrandolph The original condition is correct as we need to return empty if either group alone is provided as null or both group and item parameters provided as null.
For e.g. (consider results.Item1 is indexOfGroup and results.Item2 is indexOfItem):
Scroll with no item but group will have values for indexOfGroup = 2 and indexOfItem = -1.
Scroll with no item no group will have values for indexOfGroup = -1

The above condition functionally correct for all given use-cases but I agree that it may be confusing for readability purpose due to 'Or' operator usage instead of if-else statements. The if-else version of same code will look like:
```
if (results.Item1 == -1)
 return;
else (results.Item1 == -1 && results.Item2 == -1)
 return;
```

Hence, please let me know if I should change it to if-else for better visibility.

Reply:
else condition will never be hit, since it relies on the same condition as the if clause
from your explanation, you don't need the else clause